torch/csrc/autograd/profiler.h

```diff
@@ -131,15 +131,50 @@ struct TORCH_API Event final {
       uint16_t thread_id,
       bool record_cuda,
       at::RecordFunctionHandle handle = 0,
-      std::vector<std::vector<int64_t>>&& shapes = {})
+      std::vector<std::vector<int64_t>>&& shapes = {}, int node_id = -1)
       : name_(std::move(name)),
         kind_(kind),
         thread_id_(thread_id),
         handle_(handle),
-        shapes_(shapes) {
+        shapes_(shapes), node_id_(node_id) {
     record(record_cuda);
   }
 
+  void setNodeId(int node_id) {
+    node_id_ = node_id;
+  }
+
+  // Returns IValues corresponding to event structure, to be used for
+  // serialization.
+  at::IValue getEventIValues() const {
+    c10::impl::GenericList eventIValueList(at::AnyType::get());
+    eventIValueList.push_back(at::IValue(static_cast<int64_t>(kind_)));
+    eventIValueList.push_back(at::IValue(std::string(name_.str())));
+    eventIValueList.push_back(at::IValue(thread_id_));
+    eventIValueList.push_back(at::IValue(static_cast<double>(handle_)));
+    eventIValueList.push_back(at::IValue(node_id_));
+    return at::IValue(eventIValueList);
```

Comment:
I think we're missing other important things here, otherwise deserialized events are not useful:
shapes_, cpu_memory_usage_, cuda_memory_usage_, device_

re. cpu_ns_ - we could compute difference between start_event's cpu_ns_ and this event's cpu_ns_ and save it in Event, and serialize, parse_cpu_trace should be aware of a new field

re. CUDA - that's a bit trickier but can be handled similarly: when we're done with running profiler in remote node, we can do things similar to what we do in parse_cpu_trace - compute adjusted_time and save it in event, then serialize and send
then we only need to check in parse_cpu_trace whether we already have adjusted_time cuda time available, this should be enough to compute stats

let's discuss how to take care of the offset for outputting trace (otherwise all remote events will be start at zero in the trace)